### PR TITLE
Fix/savings app fixes

### PR DESCRIPTION
**Pull request by Kayanski**
This PR aims at providing necessary tooling for the abstract:savings-app.

Detailed changes : 

1. Added authz querier + fee granter change (https://github.com/AbstractSDK/cw-orchestrator/pull/305)
2. Authz queries implementation + tests (to know whether authz was granted)
3. Change the wasm getter implementation to get the right artifact when deploying the dex adapter. (See this comment : https://github.com/AbstractSDK/cw-orchestrator/commit/555f2799906bb9dc6adcf9dff52d1b249f19ee4e#comments)

### CODE REVIEW

File: packages/cw-orch-core/src/contract/paths.rs
Comment (CyberHoward) [inline packages/cw-orch-core/src/contract/paths.rs:1]: How are these changes related to this PR?

File: packages/cw-orch-mock/src/state.rs [hunk 1]
```diff
@@ -21,8 +23,26 @@ impl MockState {
         Self {
             addresses: HashMap::new(),
             code_ids: HashMap::new(),
+            chain_id: mock_env().block.chain_id,
```
Comment (CyberHoward) [inline packages/cw-orch-mock/src/state.rs:26]: chain-id is set in `cw_multi_test::App`, how does this relate to that? 

This could (and probably will) cause issues in the future

File: packages/cw-orch-core/src/contract/paths.rs
Comment (Kayanski) [reply packages/cw-orch-core/src/contract/paths.rs:1]: This : 
https://github.com/AbstractSDK/cw-orchestrator/commit/555f2799906bb9dc6adcf9dff52d1b249f19ee4e#comments

File: packages/cw-orch-mock/src/state.rs (hunk 1 above)
Comment (Kayanski) [reply packages/cw-orch-mock/src/state.rs:26]: Yes, however, the Deploy Details trait necessitates the chain_id field, see below in the same file. It's not a perfect solution, yes !

Comment (CyberHoward) [reply packages/cw-orch-mock/src/state.rs:26]: So for this we discussed removing this chain_id and updating the `deploy_info` function to use a seperate trait (like `EnvironmentInfo`) that we implement for each env and add as a traitbound to `CwEnv`.

**Comment by Kayanski** [comment]
This was closed by mistake because of a dumb change of branch to merge into. 
The corresponding PR is here : https://github.com/AbstractSDK/cw-orchestrator/pull/322
